### ISSUE TITLE
Use maven-shade plugin to eventually resolve classloading issue?

### ISSUE THREAD

**Issue by skuzzle** [skuzzle/restrict-imports-enforcer-rule#38]
The following pom snippet relocates all guava classes into a new package. That would resolve classloading issues as in #20 or #21.
However, this grows the final jar's size from 70kb to 2800kb which are quite some numbers.

```xml
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-shade-plugin</artifactId>
                <version>3.2.1</version>
                <configuration>
                    <artifactSet>
                        <includes>
                            <include>com.google.guava:guava</include>
                        </includes>
                    </artifactSet>
                    <relocations>
                        <relocation>
                            <pattern>com.google.common</pattern>
                            <shadedPattern>com.shaded.google.common</shadedPattern>
                        </relocation>
                    </relocations>
                </configuration>
                <executions>
                    <execution>
                        <phase>package</phase>
                        <goals>
                            <goal>shade</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
```

**Comment by skuzzle** [comment]
Might be a good idea to revisit whether it makes sense to shade/shadow the dependencies now that there is a Gradle plugin

**Comment by skuzzle** [comment]
Will be shipped with 2.5.0
